### PR TITLE
US2408: Adding AttributeService and supporting functionality to get l…

### PR DESCRIPTION
**Pull request by charlie-retzler**
…ist of AttributeTypes. Refactored some old skills stuff to use ContactAttribute class

### CODE REVIEW

File: Gateway/MinistryPlatform.Models/AttributeType.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Security.Policy;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (andycanterbury) [inline Gateway/MinistryPlatform.Models/AttributeType.cs]: Are you using threading in this class definition? I'd use ReSharper to remove unneeded references.

File: Gateway/MinistryPlatform.Models/ContactAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (andycanterbury) [inline Gateway/MinistryPlatform.Models/ContactAttribute.cs]: Clean up unneeded references here too.

File: Gateway/MinistryPlatform.Models/ContactAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace MinistryPlatform.Models
+{
+    public class ContactAttribute
+    {
+        public int dp_RecordID { get; set; }
+        public string dp_RecordName { get; set; }
+        public int dp_Selected { get; set; }
+        public int? dp_FileID { get; set; }
```
Comment (andycanterbury) [inline Gateway/MinistryPlatform.Models/ContactAttribute.cs]: do you need dp_FileID? MP always returns this and the other dp_ fields but I don't know that we ever use many besides RecordID.

File: Gateway/MinistryPlatform.Translation/Services/AttributeService.cs [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            
+        }
+
+        public List<AttributeType> GetAttributeTypes(int? attributeTypeId)
+        {
+            var token = base.ApiLogin();
+
+            var filter = ",," + attributeTypeId;
+
+            var pageViewId = Convert.ToInt32(ConfigurationManager.AppSettings["AttributesPageView"]);
```
Comment (andycanterbury) [inline Gateway/MinistryPlatform.Translation/Services/AttributeService.cs]: You could also do this with configurationWrapper.GetConfigIntValue("AttributesPageView") not sure which is better. We should standardize how we get app settings...

File: Gateway/MinistryPlatform.Translation/Services/AttributeService.cs [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            : base(authenticationService, configurationWrapper)
+        {
+            
+        }
+
+        public List<AttributeType> GetAttributeTypes(int? attributeTypeId)
+        {
+            var token = base.ApiLogin();
+
+            var filter = ",," + attributeTypeId;
```
Comment (IngageKrizJ) [inline Gateway/MinistryPlatform.Translation/Services/AttributeService.cs]: You should enclose this in quotes, like `var filter = string.format(",,\"{0}\"", attributeTypeId);`

File: Gateway/MinistryPlatform.Models/ContactAttribute.cs (hunk 3 above)
Comment (charlie-retzler) [inline Gateway/MinistryPlatform.Models/ContactAttribute.cs]: This is going away, but was old code renamed from Attribute to ContractAttribute

File: Gateway/crds-angular/Controllers/API/AttributeTypeController.cs [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+            _attributeService = attributeService;
+        }
+
+        [ResponseType(typeof (List<AttributeType>))]
+        [Route("api/AttributeType/{attributeTypeId:int?}")]
+        public IHttpActionResult Get(int? attributeTypeId = null)
+        {
+            return Authorized(token =>
+            {                
+                var attributeTypes = _attributeService.GetAttributeTypes(attributeTypeId);
```
Comment (IngageKrizJ) [inline Gateway/crds-angular/Controllers/API/AttributeTypeController.cs]: Probably overkill in this scenario, but we've typically been creating a Service (and Interface) in crds_angular project, and have separate DTO objects in that layer.  That typically does a mapping between the MinistryPlatform.Models object to an object in crds_angular, and that model is sent to the client.  That way the client isn't directly tied to the MP model.

**Pull request merged**